gui/src/2D/controls/inputText.ts

```diff
@@ -381,10 +492,105 @@ export class InputText extends Control implements IFocusableControl {
             }
         }
     }
-
     /** @hidden */
-    public processKeyboard(evt: KeyboardEvent): void {
-        this.processKey(evt.keyCode, evt.key, evt);
+    private _processDblClick(evt: PointerEvent) {
+        //pre-find the start and end index of the word under cursor, speeds up the rendering
+        this._startHighlightIndex = this._text.length - this._cursorOffset;
+        this._endHighlightIndex = this._startHighlightIndex;
+        for (let rWord = /\w+/g, left = 1, right = 1; this._startHighlightIndex > 0 && this._endHighlightIndex < this._text.length && (left || right);) {
+            right = (this._text[this._endHighlightIndex].search(rWord) !== -1) ? ++this._endHighlightIndex : 0;
+            left =  (this._text[this._startHighlightIndex - 1 ].search(rWord) !== -1) ? --this._startHighlightIndex : 0;
+        }
+        this._isTextHighlightOn = true;
+        this._blinkIsEven = false;
+    }
+
+    /**
+     * Called by the onClipboardObservable of AdvanceDynamicTexture
+     * @param evt Event object
+     */
+    public processKeyboard(evt: KeyboardEvent | PointerEvent | ClipboardEvent): void {
+        // check the event type and call its action
+        if (evt instanceof KeyboardEvent) {
```

Comment:
These casts will become unneeded with my  proposed changed. (It is always a good hint that something is not going well when you need to test types)